From 8dbc47fc7a449d6ec1eb02a80c595f8dc0ed738a Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 30 Apr 2015 13:41:43 +0800
Subject: [PATCH] MLK-10785 mxc ipuv3 fb: Correct ->resolve in bailout of
 ->set_par due to no PRE

We should set mxcfbi->resolve to be false in the bailout path of ops->set_par
due to no PRE is available.  Otherwise, the users may call the PAN_DISPLAY
ioctrl to switch buffer right after the bailout path and accidently runs the
resolving logic for switching buffer.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 drivers/video/mxc/mxc_ipuv3_fb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/video/mxc/mxc_ipuv3_fb.c b/drivers/video/mxc/mxc_ipuv3_fb.c
index aa13b02..c303404 100644
--- a/drivers/video/mxc/mxc_ipuv3_fb.c
+++ b/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -567,6 +567,7 @@ static int _setup_disp_channel2(struct fb_info *fbi)
 				dev_err(fbi->device,
 					"failed to alloc PRE\n");
 				mxc_fbi->prefetch = mxc_fbi->cur_prefetch;
+				mxc_fbi->resolve = false;
 				return mxc_fbi->pre_num;
 			}
 		}
-- 
1.8.0

